pandas/core/generic.py

```diff
@@ -129,6 +130,34 @@ def __init__(self, data, axes=None, copy=False, dtype=None,
         object.__setattr__(self, '_data', data)
         object.__setattr__(self, '_item_cache', {})
 
+    def _ipython_display_(self):
+        try:
+            from IPython.display import display
+        except ImportError:
+            return None
+
+        # Series doesn't define _repr_html_ or _repr_latex_
+        latex = self._repr_latex_() if hasattr(self, '_repr_latex_') else None
+        html = self._repr_html_() if hasattr(self, '_repr_html_') else None
+        table_schema = self._repr_table_schema_()
```

Comment:
Should the `config.get_option` check happen here so it doesn't end up as:

```
"application/vnd.dataresource+json": None
```

in the resulting output?

Oh nevermind, I see the `if v` in the dict comprehension.